healthx/handler.go

```diff
@@ -79,17 +79,18 @@ func NewHandler(
 
 type router interface {
 	GET(path string, handle httprouter.Handle)
+	Handler(method, path string, handler http.Handler)
 }
 
 // SetHealthRoutes registers this handler's routes for health checking.
-func (h *Handler) SetHealthRoutes(r router, shareErrors bool) {
-	r.GET(AliveCheckPath, h.Alive)
-	r.GET(ReadyCheckPath, h.Ready(shareErrors))
+func (h *Handler) SetHealthRoutes(r router, shareErrors bool, middleware func(http.Handler) http.Handler) {
+	r.Handler("GET", AliveCheckPath, middleware(h.Alive()))
+	r.Handler("GET", ReadyCheckPath, middleware(h.Ready(shareErrors)))
```

Comment:
In most cases we don't want to set the middleware. I suggest we use the options pattern here instead.
```suggestion
func (h *Handler) SetHealthRoutes(r router, shareErrors bool, opts ...Options) {
	o := getOptions(opts)
	r.Handler("GET", AliveCheckPath, o.middleware(h.Alive()))
	r.Handler("GET", ReadyCheckPath, o.middleware(h.Ready(shareErrors)))
```
Per default the middleware would be a noop then.

Reply:
By that we also avoid introducing a breaking change.

Reply:
Thanks for the tip, will update this

Reply:
Though, I was thinking, more like this.

if it makes sense

```go
type Middleware func(http.Handler) http.Handler

// SetHealthRoutes registers this handler's routes for health checking.
func (h *Handler) SetHealthRoutes(r router, shareErrors bool, opts ...Middleware) {
	aliveHandler := h.Alive()
	readyHandler := h.Ready(shareErrors)

	for _, opt := range opts {
		aliveHandler = opt(aliveHandler)
	}

	for _, opt := range opts {
		readyHandler = opt(readyHandler)
	}

	r.Handler("GET", AliveCheckPath, aliveHandler)
	r.Handler("GET", ReadyCheckPath, readyHandler)
}
```